Source/WebCore/platform/gstreamer/GStreamerQuirkBroadcomBase.cpp

```diff
@@ -0,0 +1,200 @@
+
+    if (correctedBufferingPercentage2 >= 100)
+        state.m_streamBufferingLevelMovingAverage.reset(100);
+    int averagedBufferingPercentage = state.m_streamBufferingLevelMovingAverage.accumulate(correctedBufferingPercentage2);
+
+#ifndef GST_DISABLE_GST_DEBUG
+    const char* extraElements = state.m_multiqueue ? "playpump and multiqueue" : "playpump";
+    if (!originalBufferingPercentage) {
+        GST_DEBUG("[Buffering] Buffering: mode: GST_BUFFERING_STREAM, status: %d%% (corrected to %d%% with current-level-bytes, "
+            "to %d%% with %s content, and to %d%% with moving average).", originalBufferingPercentage, correctedBufferingPercentage1,
+            correctedBufferingPercentage2, extraElements, averagedBufferingPercentage);
+    } else {
+        GST_DEBUG("[Buffering] Buffering: mode: GST_BUFFERING_STREAM, status: %d%% (corrected to %d%% with %s content and "
+            "to %d%% with moving average).", originalBufferingPercentage, correctedBufferingPercentage2, extraElements,
+            averagedBufferingPercentage);
+    }
+#endif
+
+    return averagedBufferingPercentage;
+}
+
+void GStreamerQuirkBroadcomBase::resetBufferingPercentage(MediaPlayerPrivateGStreamer* playerPrivate, int bufferingPercentage) const
+{
+    auto state = ensureState(playerPrivate);
+    state.m_streamBufferingLevelMovingAverage.reset(bufferingPercentage);
+}
+
+void GStreamerQuirkBroadcomBase::setupBufferingPercentageCorrection(MediaPlayerPrivateGStreamer* playerPrivate, GstState currentState, GstState newState, GstElement* element) const
+{
+    auto state = ensureState(playerPrivate);
+
+    // This code must support being run from different GStreamerQuirkBroadcomBase subclasses without breaking. Only the
+    // first subclass instance should run.
+
+    if (currentState == GST_STATE_NULL && newState == GST_STATE_READY) {
+        bool alsoGetMultiqueue = false;
+        if (!g_strcmp0(G_OBJECT_TYPE_NAME(element), "GstBrcmVidFilter")) {
+            state.m_vidfilter = element;
+            alsoGetMultiqueue = true;
+        } else if (!g_strcmp0(G_OBJECT_TYPE_NAME(element), "GstBrcmAudFilter")) {
+            state.m_audfilter = element;
+            alsoGetMultiqueue = true;
+        } else if (!g_strcmp0(G_OBJECT_TYPE_NAME(element), "GstQueue2"))
+            state.m_queue2 = element;
+
+        // Might have been already retrieved by vidfilter or audfilter, whichever appeared first.
+        if (alsoGetMultiqueue && !state.m_multiqueue) {
+            // Also get the multiqueue (if there's one) attached to the vidfilter/aacparse+audfilter.
+            // We'll need it later to correct the buffering level.
+            for (auto* sinkPad : GstIteratorAdaptor<GstPad>(GUniquePtr<GstIterator>(gst_element_iterate_sink_pads(element)))) {
+                GRefPtr<GstPad> peerSrcPad = adoptGRef(gst_pad_get_peer(sinkPad));
+                if (peerSrcPad) {
```

Comment:
In this specific case, where the loop only has one possible iteration, `continue` would work as intended (the loop would exit because there aren't more pending iterations).